### PR TITLE
Bosch BMI055 & BMI088 don't publish duplicated reads

### PR DESCRIPTION
**Pull request by dagar**

### CODE REVIEW

**Comment by dagar** [comment]
@ryankierteal you might want to give this a try. Possibly related to https://github.com/PX4/ecl/issues/777.

**Comment by dagar** [comment]
This doesn't fix the fundamental accel + gyro synchronization issues at the sensors hub, but it's definitely less bad than what's currently in master.

Post v1.11 release I'm going to pursue 2 things to properly ensure we have synchronized IMU for the estimator.
1. move actual IMU integration to `sensors/vehicle_imu` where we can go out of our way to ensure data is synchronized over the integration period.
2. Finish the new Bosch BMI055 & BMI088 drivers that use the sensor FIFOs and data ready interrupts. The actual sensor driver output (publication) will then be much more regular and easier to synchronize downstream.

File: src/drivers/imu/bmi055/BMI055_gyro.cpp [hunk 1]
```diff
@@ -360,6 +362,18 @@ BMI055_gyro::RunImpl()
 		return;
 	}
 
+	// don't publish duplicated reads
+	if ((report.gyro_x == _gyro_prev[0]) && (report.gyro_y == _gyro_prev[1]) && (report.gyro_z == _gyro_prev[2])) {
+		perf_end(_sample_perf);
+		perf_count(_duplicates);
+		return;
+
+	} else {
```
Comment (bkueng) [inline src/drivers/imu/bmi055/BMI055_gyro.cpp:371]: No `else` needed, since you have the `return`.

Comment (dagar) [reply src/drivers/imu/bmi055/BMI055_gyro.cpp:371]: Thanks, I was just trying to keep the comparison together as a single block.

**Pull request merged**